### PR TITLE
[code] use enum class and some constexpr

### PR DESCRIPTION
**Pull request by gocarlos**
* using enum class to prevent to prevent conflicts with C stuff
* adding constexpr because why not

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -51,6 +51,7 @@ build
 #tools
 .vscode
 .idea
+CMakeLists.txt.user
```
Comment (wgtdkp) [inline .gitignore]: Is this necessary?

File: src/library/commissioner_impl.cpp [hunk 2]
```diff
@@ -213,7 +213,8 @@ Error CommissionerImpl::ValidateConfig(const Config &aConfig)
         VerifyOrExit(commissionerIdTlv.IsValid(), error = "invalid commissioner ID: " + aConfig.mId);
     }
 
-    VerifyOrExit(aConfig.mLogLevel <= 5, error = "invalid logging level: " + std::to_string(aConfig.mLogLevel));
+    VerifyOrExit(aConfig.mLogLevel <= LogLevel::kDebug,
+                 error = "invalid logging level: " + std::to_string(static_cast<int>(aConfig.mLogLevel)));
```
Comment (wgtdkp) [inline src/library/commissioner_impl.cpp]:
```suggestion
                 error = "invalid logging level: " + std::to_string(utils::to_underlying(aConfig.mLogLevel)));
```

File: include/commissioner/border_agent.hpp [hunk 3]
```diff
@@ -136,21 +136,21 @@ struct BorderAgent
...
-    static const uint16_t kVendorNameBit      = 1 << 6;
-    static const uint16_t kModelNameBit       = 1 << 7;
-    static const uint16_t kActiveTimestampBit = 1 << 8;
-    static const uint16_t kPartitionIdBit     = 1 << 9;
-    static const uint16_t kVendorDataBit      = 1 << 10;
-    static const uint16_t kVendorOuiBit       = 1 << 11;
-    static const uint16_t kDomainNameBit      = 1 << 12;
-    static const uint16_t kBbrSeqNumberBit    = 1 << 13;
-    static const uint16_t kBbrPortBit         = 1 << 14;
+    static constexpr uint16_t kAddrBit            = 1 << 0;
```
Comment (wgtdkp) [inline include/commissioner/border_agent.hpp]: I like this. How about we do this also to constants in [network_data.hpp](https://github.com/openthread/ot-commissioner/blob/master/include/commissioner/network_data.hpp)?

File: .gitignore (hunk 1 above)
Comment (gocarlos) [reply .gitignore]: yes, I work with qtcreator, this this the pendent to .vscode

**Comment by gocarlos** [comment]
poh, I see that each travis ci job is compiling cmake?

**Comment by gocarlos** [comment]
is it an option to do a `pip3 install cmake` to get the newer version of cmake?

**Comment by wgtdkp** [comment]
> is it an option to do a `pip3 install cmake` to get the newer version of cmake?

Yes, I think it should work.

**Comment by wgtdkp** [comment]
@gocarlos The travis tests failed on style checking. You need to run
```shell
./script/make-pretty.sh
```
before submitting.

**Pull request merged**